eventgrid/data-plane/src/main/java/com/microsoft/azure/eventgrid/customization/EventGridSubscriber.java

```diff
@@ -40,114 +44,124 @@ public EventGridSubscriber() {
 
     /**
      * Add a custom event mapping. If a mapping with same eventType exists then the old eventDataType is replaced by
-     * the specified eventDataType
+     * the specified eventDataType.
      *
-     * @param eventType the event type name
-     * @param eventDataType type of the Java model that the event type name mapped to
+     * @param eventType the event type name.
+     * @param eventDataType type of the Java model that the event type name mapped to.
      */
     @Beta
-    public void putCustomEventMapping(String eventType, Type eventDataType) {
+    public void putCustomEventMapping(final String eventType, final Type eventDataType) {
         if (eventType == null || eventType.isEmpty()) {
             throw new IllegalArgumentException("eventType parameter is required and cannot be null or empty");
         }
         if (eventDataType == null) {
             throw new IllegalArgumentException("eventDataType parameter is required and cannot be null");
         }
-        this.eventTypeToEventDataMapping.put(eventType.toLowerCase(), eventDataType);
+        this.eventTypeToEventDataMapping.put(canonicalizeEventType(eventType), eventDataType);
     }
 
     /**
      * Get type of the Java model that is mapped to the given eventType.
      *
-     * @param eventType the event type name
-     * @return type of the Java model id mapping exists, null otherwise
+     * @param eventType the event type name.
+     * @return type of the Java model if mapping exists, null otherwise.
      */
     @Beta
-    public Type getCustomEventMapping(String eventType) {
-        if (eventType == null || eventType.isEmpty()) {
-            throw new IllegalArgumentException("eventType parameter is required and cannot be null or empty");
-        }
-        if (!this.eventTypeToEventDataMapping.containsKey(eventType.toLowerCase())) {
+    public Type getCustomEventMapping(final String eventType) {
+        if (!containsCustomEventMappingFor(eventType)) {
             return null;
+        } else {
+            return this.eventTypeToEventDataMapping.get(canonicalizeEventType(eventType));
         }
-        return this.eventTypeToEventDataMapping.get(eventType.toLowerCase());
+    }
+
+    /**
+     * @return get all registered custom event mappings.
+     */
+    @Beta
+    public Set<Map.Entry<String, Type>> getAllCustomEventMapping() {
```

Comment:
Nit: Should this have a "s" at the end to denote the plurality (getAllCustomEventMappings())?

Reply:
thanks for catching the typo, will fix it.